maven-artifact/src/main/java/org/apache/maven/artifact/versioning/ComparableVersion.java

```diff
@@ -40,22 +41,39 @@
  *     <code>1.0alpha1 =&gt; [1, 0, alpha, 1]</code></li>
  * <li>unlimited number of version components,</li>
  * <li>version components in the text can be digits or strings,</li>
- * <li>strings are checked for well-known qualifiers and the qualifier ordering is used for version ordering.
- *     Well-known qualifiers (case insensitive) are:<ul>
- *     <li><code>alpha</code> or <code>a</code></li>
- *     <li><code>beta</code> or <code>b</code></li>
- *     <li><code>milestone</code> or <code>m</code></li>
- *     <li><code>rc</code> or <code>cr</code></li>
- *     <li><code>snapshot</code></li>
- *     <li><code>(the empty string)</code> or <code>ga</code> or <code>final</code></li>
- *     <li><code>sp</code></li>
- *     </ul>
- *     Unknown qualifiers are considered after known qualifiers, with lexical order (always case insensitive),
- *   </li>
- * <li>a hyphen usually precedes a qualifier, and is always less important than something preceded with a dot.</li>
+ * <li>
+ *   String qualifiers are ordered lexically (case insensitive), with the following exceptions:
+ *   <ul>
+ *     <li> 'snapshot' &lt; '' &lt; 'sp' </li>
```

Comment:
`sp` for service pack?

Reply:
indeed. SP has been hard coded on maven code since ages. the maven docs now discourages its use. but removing it altogether might not be without surprises. most known are Hibernate and Weld using 'Final' qualifiers instead of none, and Weld using SP1 instead of patch versions increment. i hope to see them change their release procedures in the future.

Reply:
Maven team might want at some point to not support SP qualifiers anymore. in that case i will happily change the code and documentation. letting as is for now.

Reply:
So what about really following semantic version instead of creating exceptions?

Reply:
> So what about really following semantic version instead of creating exceptions?

A breaking change suitable for the Major 4 update. if the Maven team accepts, i can perform a separate PR removing the exceptions in the master branch only.

Reply:
> > So what about really following semantic version instead of creating exceptions?
> 
> A breaking change suitable for the Major 4 update. if the Maven team accepts, i can perform a separate Jira/PR removing the exceptions in the master branch only.

I agree with that. Please, let's leave breaking changes out of this PR and not conflate things.

Reply:
Michael, could you perhaps write a post to the mailing list with a vote about that if you agree? Thanks!

Reply:
The cleanup you mean?

Reply:
> The cleanup you mean?

i think Andrzej meant a vote for Maven 4 to be semver compatible or at least move towards it step by step (PR by PR)